mentat/sampler/sample.py

```diff
@@ -0,0 +1,268 @@
+                pass
+            stream.send(
+                f"Found repo URL: {remote_url}. Press 'ENTER' to accept, or enter a new"
+                " URL."
+            )
+            response = (await collect_user_input()).data.strip()
+            if response == "y":
+                repo = remote_url
+            else:
+                repo = response
+            config.sample_repo = repo
+
+        stream.send("Sample Title:")
+        title = (await collect_user_input()).data.strip() or ""
+        stream.send("Description: (optional)")
+        description = (await collect_user_input()).data.strip() or ""
+        stream.send("Test Command: (optional, e.g. 'pytest -k foo')")
+        test_command = (await collect_user_input()).data.strip() or ""
+
+        messages = list[dict[str, str]]()
+        for m in conversation.get_messages():
+            parsed = parse_message(m)
+            if parsed and m["role"] in {"user", "assistant"}:
+                # TODO Replace mentat-formatted edits with git diffs
+                messages.append({"role": m["role"], "content": parsed})
+                # TODO Remove edits altogether from last assistant message
+
+        args = list[str]()
+        if code_context.include_files:
+            feature_refs = get_consolidated_feature_refs(
+                [f for fs in code_context.include_files.values() for f in fs]
+            )
+            args += [get_relative_path(Path(f), cwd).as_posix() for f in feature_refs]
+
+        return cls(
+            title=title,
+            description=description,
+            id=uuid4().hex,
+            parent_id=code_file_manager.history.last_sample_id or "",
+            repo=repo,
+            merge_base=code_file_manager.history.merge_base,
+            diff_merge_base=code_file_manager.history.diff_merge_base or "",
+            diff_active=code_file_manager.history.diff_active or "",
+            messages=messages,
+            args=args,
+            diff_edit=get_diff_active() or "",  # TODO: subtract diff_active
+            test_command=test_command,
+        )
+
+    def save(self, fname: str) -> None:
+        with open(fname, "w") as f:
+            json.dump(attr.asdict(self), f, indent=4)
+
+    @classmethod
+    def load(cls, fname: str) -> Sample:
+        with open(fname, "r") as f:
+            return cls(**json.loads(f.read()))
+
+    async def eval(self, path_to_repo: Path | str | None = None) -> dict[str, str]:
+        cwd = setup_repo(self, path_to_repo)
+        await run_mentat_on_sample(self, cwd)
+
+        # EVALUATE
+        diff_eval = get_diff_active() or ""  # TODO: subtract diff_active
+
+        # Run the test command
+        test_result: str = ""
+        if self.test_command:
+            test_result = subprocess.check_output(
+                self.test_command,
+                text=True,
+            )
+        # TODO: Run the LLM evaluations from benchmark
+        # TODO: Save the results
```

Comment:
I assume these TODOs are for another PR? Doesn't seem necessary to nail down how samples will turned into benchmarks in this one.